lang/dart.md

```diff
@@ -0,0 +1,189 @@
+
+## What is Dart used for?
+Dart programming language can be used for web and mobile applications and is also suitable for server and desktop applications.
+Dart wasn’t popular and was known as an orphan, but it had a compiler for a Dart VM, also Google had all the control about this language.
+
+**Client Web App Frameworks:**
+- **[AngularDart](https://github.com/angulardart/angular)** - AngularDart is a development platform for building mobile and desktop web applications.
+- **[Flutter](https://flutter.dev/)** - Flutter is a framework to build high-performance, cross-platform mobile apps, allowing applications to be written for Android, iOS and Web Apps.
+- **[MDL/Dart](http://mdl.mikemitterer.at)** - Material Design Lite for Dart is a framework of components for web developers based on Google's Material Design philosophy.
+- **[OverReact](https://workiva.github.io/over_react)** - A library for building statically-typed React UI components.
+- **[VueDart](https://vuedart.dev)** - Create Vue web apps using Dart, a progressive, incrementally-adoptable framework for building Web UIs.
+
+**Server Frameworks:**
+- **[Jaguar](https://github.com/Jaguar-dart/jaguar)** - A server framework built for speed, simplicity and extensibility.
+- **[Start](https://github.com/lvivski/start)** - Sinatra inspired web framework to serve static files, handle dynamic requests, websockets and create JSON responses.
+- **[Shelf](https://pub.dev/packages/shelf)** - Shelf makes it easy to create and compose web servers and parts of web servers.
+- **[Vane](https://github.com/Scorpiion/Vane)** - Framework with built-in server runtime environment and middleware system.
+- **[Rikulo Stream](https://github.com/rikulo/stream)** - Lightweight web server with request routing, filtering, template engine, WebSocket, MVC design pattern, and file-based static resources.
+- **[Alfred]( https://github.com/rknell/alfred)** - Minimal ExpressJS-inspired server with routing and middlware.
+ 
+## Why to use Dart?
+- Dart is really flexible
+- Dart embraced the open source ecosystem
+- Dart is pretty easy to learn
+- Dart is a good first language
+- Dart has great tooling support
+- Dart is a robust language
+- Dart ensures productivity
+- Dart is used by Flutter
+- Dart is backed by a tech giant
+- Dart solved a lot of problems, efficiently
+
+## Tools You Will Need
+For setting up environment for performing java codes you need with a minimum of 2 GB of RAM.
+You will also need the following softwares :
+- Linux, macOS or Windows xp/7/8/10 operating system
+- Dart SDK
+- Text Editors
```

Comment:
What kind of Text Editors, beginners will use? :thinking: 
Pl mention! : )

Reply:
Most people will use:
1. Sublime Text
2. Visual Studio Code
3. Notepad++
4. Atom

Reply:
@chandan0000: Yep, agree with Vasanth, did you give a check to include such stuff with more links to reference?